projects/laji/src/app/+project-form/form/named-place/np-choose/np-list/np-list.component.ts

```diff
@@ -165,27 +172,42 @@ export class NpListComponent implements OnDestroy {
       && this._fields.indexOf('$.reserve.reserver') === -1) {
       this._fields.push('$.reserve.reserver');
     }
-    const cols: ObservationTableColumn[] = [];
-    for (const path of this._fields) {
-      const {cellTemplate, ...columnMetadata} = this.columnsMetaData[path] || {} as DatatableColumn;
-      const col: DatatableColumn = {
+
+
+    const cols = this._fields.reduce((_cols, path) => {
+
+      const splits = path.split('.');
```

Comment:
It looks to me that this does about the same than the lines 87-97 in the `np-info.component` but maybe there are differences? I think it would be clearer if all the schema parsing stuff was moved to some separate service since they make the functions long and little hard to read. This would become `const schema = someService.getSchemaFromPath(path)`

Reply:
Good idea, refactored the schema parsing into a single static function.